util/EIP-2124/src/lib.rs

```diff
@@ -95,77 +96,126 @@ pub struct ForkId {
 pub enum RejectReason {
 	/// Remote node is outdated and needs a software update.
 	RemoteStale,
-	/// Local node is on an incompatible chain or needs a sofwtare update.
+	/// Local node is on an incompatible chain or needs a software update.
 	LocalIncompatibleOrStale,
 }
 
 /// Filter that describes the state of blockchain and can be used to check incoming `ForkId`s for compatibility.
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, MallocSizeOf)]
+#[cfg_attr(any(test, feature = "test-helpers"), derive(PartialEq))]
```

Comment:
I'm not sure deriving `PartialEq` warrants a test feature especially given the cargo bug that features leak from dev- and build-dependencies (https://github.com/rust-lang/cargo/issues/4866). We might as well do this unconditionally.

Reply:
I'm very hesitant about allowing users to compare private internal state. If you insist I can expose this implementation though.

Reply:
done in `ethereum-forkid` crate